core/src/main/scala/org/broadinstitute/dsde/rawls/user/UserService.scala

```diff
@@ -308,6 +312,35 @@ class UserService(protected val userInfo: UserInfo, val dataSource: SlickDataSou
     gcsDAO.deleteProject(projectName).map(_ => RequestComplete(StatusCodes.OK))
   }
 
+  def registerBillingProject(projectName: RawlsBillingProjectName,
+                             owner: RawlsUserEmail,
+                             bucket: String): Future[PerRequestMessage] = {
+    val registrationAttempt =
+      for {
+        _ <- samDAO.createResource(SamResourceTypeNames.billingProject, projectName.value, userInfo)
+        _ <- samDAO.overwritePolicy(SamResourceTypeNames.billingProject, projectName.value, workspaceCreatorPolicyName, SamPolicy(Seq.empty, Seq.empty, Seq(SamProjectRoles.workspaceCreator)), userInfo)
+        _ <- samDAO.overwritePolicy(SamResourceTypeNames.billingProject, projectName.value, canComputeUserPolicyName, SamPolicy(Seq.empty, Seq.empty, Seq(SamProjectRoles.batchComputeUser, SamProjectRoles.notebookUser)), userInfo)
+        project <- dataSource.inTransaction { dataAccess =>
+          dataAccess.rawlsBillingProjectQuery.create(
+            RawlsBillingProject(
+              projectName,
+              RawlsGroup(
+                RawlsGroupName(SamProjectRoles.owner),
+                RawlsGroupEmail("will be removed after rebasing"),
+                Set.empty,
+                Set.empty),
+              bucket,
+              CreationStatuses.Ready,
+              None,
+              None))
+        }
+      } yield project
+
+    registrationAttempt recoverWith {
+      case t: Throwable => Future.successful(RequestComplete(ErrorReport(StatusCodes.InternalServerError, t.getMessage)))
+    } map { _ => RequestComplete(StatusCodes.Created) }
+  }
+
```

Comment:
Is this what you had in mind @dvoet?

Reply:
yes, much nicer. You don't need the recover. It is actually detrimental as it covers up details of the original error.